packages/eslint-plugin/src/rules/consistent-type-exports.ts

```diff
@@ -84,6 +84,7 @@ export default util.createRule<Options, MessageIds>({
       const symbol = services.getSymbolAtLocation(specifier.exported);
       const aliasedSymbol = checker.getAliasedSymbol(symbol!);
 
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-enum-comparison
       if (!aliasedSymbol || aliasedSymbol.escapedName === 'unknown') {
```

Comment:
Lots of complaints from `no-unsafe-enum-comparison`, and I think there's a bug in the rule... ran out of time today to investigate though.

Reply:
Just to be clear, I think this PR is still ready to review despite these annoying `eslint-disable-next-line`s. If it is a bug in the rule then I'll file an issue & send a PR resolving the bug.

Reply:
so `.escapedName` is typed as [`ts.__String`](https://github.com/microsoft/TypeScript/blob/5021dd87ef36e87bae39cbb5a30dabd295410920/src/compiler/types.ts#L5981):
```ts
export type __String =
  | (string & { __escapedIdentifier: void; })
  | (void & { __escapedIdentifier: void })
  | InternalSymbolName;
```

Note the `InternalSymbolName` type in that union - [that is an enum](https://github.com/microsoft/TypeScript/blob/5021dd87ef36e87bae39cbb5a30dabd295410920/src/compiler/types.ts#L5953-L5971)!

The rule has handling for some cases, but it doesn't appear to detect branded types. Filed https://github.com/typescript-eslint/typescript-eslint/issues/7114